File: 06Sort/6-01 선택정렬.py
arr=[7,5,9,0,3,1,6,2,4,8]
for i in range(len(arr)):
    idx=i
    for j in range(i+1,len(arr)):
        if arr[idx]>arr[j]: #idx가 j보다 크다면
            idx=j #j로 변경
    arr[i],arr[idx]=arr[idx],arr[i] #스와프
print(arr) #오름차순 정렬

arr=[7,5,9,0,3,1,6,2,4,8]
for i in range(len(arr)):
    idx=i
    for j in range(i+1,len(arr)):
        if arr[idx]<arr[j]:
            idx=j
    arr[i],arr[idx]=arr[idx],arr[i] #스와프
print(arr) #내림차순 정렬

File: 06Sort/6-01 선택정렬.cpp
#include <iostream>
using namespace std;

int n = 10;
int arr[10] = { 7, 5, 9, 0, 3, 1, 6, 2, 4, 8 };

void swap(int *a, int *b) {
    int *temp = a;
    a = b;
    b = temp;
}

int main(void) {
    for (int i = 0; i < n; i++) {
        int min_idx = i; //최소값을 i로 설정
        for (int j = i; j < n; j++) {
            if (arr[min_idx] > arr[j]) { //더 작은 값을 만나면
                min_idx = j; 
            }
        }
        swap(arr[i], arr[min_idx]); //교체
    }
    for (int i = 0; i < n; i++) {
        cout << arr[i] << ' ';
    }
}